server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2393,18 +2409,24 @@ public UserVm updateVirtualMachine(UpdateVMCmd cmd) throws ResourceUnavailableEx
         Map<String,String> details = cmd.getDetails();
         List<Long> securityGroupIdList = getSecurityGroupIdList(cmd);
         boolean cleanupDetails = cmd.isCleanupDetails();
+        String extraConfig = cmd.getExtraConfig();
 
         UserVmVO vmInstance = _vmDao.findById(cmd.getId());
 
         if (isDisplayVm != null && isDisplayVm != vmInstance.isDisplay()) {
             updateDisplayVmFlag(isDisplayVm, id, vmInstance);
         }
         if (cleanupDetails){
-            _vmDetailsDao.removeDetails(id);
-        }
-        else if (MapUtils.isNotEmpty(details)) {
-            vmInstance.setDetails(details);
-            _vmDao.saveDetails(vmInstance);
+            userVmDetailsDao.removeDetails(id);
+        } else {
+            if (MapUtils.isNotEmpty(details)) {
+                vmInstance.setDetails(details);
+                _vmDao.saveDetails(vmInstance);
+            }
+            if (!Strings.isNullOrEmpty(extraConfig)) {
+                Account account = CallContext.current().getCallingAccount();
```

Comment:
You can use accountDao at this point to get the account by ID (you can get it from vmInstance.getAccountId())